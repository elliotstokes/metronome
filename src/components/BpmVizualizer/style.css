.bpm-container {
  width: 266px;
  height: 266px;
  position: relative;
  margin: 0 auto;
}

.circle {
  border-radius: 50%;
}

.bpm-label {
  position: absolute;
  width: 50%;
  height: 50%;
  background-color: rgb(100, 188, 164);
  color: rgb(22, 16, 32);
  border-radius: 50%;
  transform: translate(50%,50%);
  font-size: 24px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.bpm-pulse {
  width: 100%;
  height: 100%;
  position: absolute;
  background-color:  rgb(100, 188, 164);
  animation-iteration-count: infinite;
  animation-name: pulse-ring;
  animation-timing-function: ease-out;
}

@keyframes pulse-ring {
  0% {
    transform: scale(.33);
  }
  80%, 100% {
    opacity: 0;
  }
}