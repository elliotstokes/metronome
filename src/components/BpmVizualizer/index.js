import React from "react"
import './style.css'
import { beatLength } from '../../config/bpm'

const BpmVizualizer = ({ bpm }) => (
    <div className={'bpm-container'}>
      <div className={'bpm-pulse circle'} style={{animationDuration: `${beatLength(bpm)}s`}} />
      <div className={'bpm-label circle'}>{bpm}</div>
    </div>
  )

export default BpmVizualizer