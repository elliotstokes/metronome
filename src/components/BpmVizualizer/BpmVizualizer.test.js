import React from 'react'
import { create } from 'react-test-renderer'
import BpmVizualizer from '.'

test('renders correctly', () => {
  const component = create(<BpmVizualizer  bpm={128} />)
  expect(component.toJSON()).toMatchSnapshot()
})