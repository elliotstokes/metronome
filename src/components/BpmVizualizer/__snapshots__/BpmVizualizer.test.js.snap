// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bpm-container"
>
  <div
    className="bpm-pulse circle"
    style={
      Object {
        "animationDuration": "0.46875s",
      }
    }
  />
  <div
    className="bpm-label circle"
  >
    128
  </div>
</div>
`;
